src/java.base/share/classes/sun/nio/ch/SimpleAsynchronousFileChannelImpl.java

```diff
@@ -181,8 +181,10 @@ public void force(boolean metaData) throws IOException {
         if (!shared && !writing)
             throw new NonWritableChannelException();
 
+        long len = size != 0 ? size : Long.MAX_VALUE - Math.max(0, position);
```

Comment:
In other places, the changes just checks if size is 0 and set it to Long.MAX_VALUE - Math.max(0, position). This avoids having both "len" and "size" and avoids potential mistakes. Can we do the same here and in the Windows implementation?

Reply:
In both cases the `size` parameter is final and cannot be reassigned, so either the parameter has to be made non-final or a new variable has to be used.

Reply:
Okay, I see it would be disruptive to do that. In that case, let's go with what you have (maybe put parentheses around the size != 0 condition in the ternary to make it a bit clearer).

Overall I think we've got the two original corner cases to a good place with minimal-to-no impact.